mev_inspect/aave_liquidations.py

```diff
@@ -0,0 +1,107 @@
+from typing import List, Optional, Dict
+
+from mev_inspect.traces import get_child_traces
+from mev_inspect.schemas.classified_traces import (
+    ClassifiedTrace,
+    DecodedCallTrace,
+    Classification,
+    Protocol,
+)
+
+from mev_inspect.schemas.liquidations import Liquidation
+
+AAVE_CONTRACT_ADDRESSES: List[str] = [
+    # AAVE Proxy
+    "0x398ec7346dcd622edc5ae82352f02be94c62d119",
+    # AAVE V2
+    "0x7d2768de32b0b80b7a3454c06bdac94a69ddc7a9",
+    # AAVE V1
+    "0x3dfd23a6c5e8bbcfc9581d2e864a68feb6a076d3",
+    # AAVE V2 WETH
+    "0x030ba81f1c18d280636f32af80b9aad02cf0854e",
+]
+
+
+def find_liquidator_payback(
+    trace: ClassifiedTrace, liquidator: str
+) -> Optional[ClassifiedTrace]:
+    """Finds liquidator payback """
+
+    if isinstance(trace, DecodedCallTrace):
+
+        if "recipient" in trace.inputs:
+
+            if (
+                trace.inputs["recipient"] == liquidator
+                and trace.from_address in AAVE_CONTRACT_ADDRESSES
+            ):
+                return trace
+
+        elif "dst" in trace.inputs:
+            if (
+                trace.inputs["dst"] == liquidator
+                and trace.from_address in AAVE_CONTRACT_ADDRESSES
+            ):
+                return trace
+
+    return None
+
+
+def get_liquidations(
+    traces: List[ClassifiedTrace],
+) -> List[Liquidation]:
+
+    """Inspect list of classified traces and identify liquidation"""
+    liquidations: List[Liquidation] = []
+
+    for trace in traces:
+
+        if trace.classification == Classification.liquidate and isinstance(
+            trace, DecodedCallTrace
+        ):
+
+            liquidator = trace.from_address
+
+            child_traces = get_child_traces(
+                trace.transaction_hash, trace.trace_address, traces
+            )
+
+            for child in child_traces:
+
+                if child.classification == Classification.liquidate:
+                    traces.remove(child)
+
+                liquidator_payback = find_liquidator_payback(child, liquidator)
+
+                if liquidator_payback:
+
+                    assert isinstance(
+                        liquidator_payback.inputs, Dict
+                    )  # Necessary for mypy to indentify type
+
+                    if "amount" in liquidator_payback.inputs:
```

Comment:
great place for a function

rather than having multiple branches which assign to `received_amount`

assign to it once
```
received_amount = get_received_amount(...)
```